### ISSUE TITLE
Object recognition with customer face reognition

### ISSUE THREAD

**Issue by akhilgupta0221** [srianant/computer_vision#12]
Hi Srianant

Now a days I am working on a "Automate retail checkout" ( i.e : amazon GO) project and here I am planning to use the Openpose.
I have a query related with the same.

------------
Use case:  |
------------
1) To identify  which product customer has picked form the shelf.

Assumption:
--------------
We already trained the algo for the objects recognition and tracking  and we know the initial coordinates of all the objects in the shelf.

Solution: 
---------
If the coordinate of the object and palm is same then we mark user has checked out the  product from the shelf.
.
Question :
-----------
How we can associate the current checked out product with a specific user,
In other words , how we can associate the checked out product with face recognition So, that we make sure the current product is checked out by a specific user and  amount can be deducted accordingly.

Regards
Akhil

**Comment by srianant** [comment]
There are many ways this fundamental problem could be addressed. Here is just one method.
- Facial landmarks (say using openpose face keypoints or DLIB) can be represented as 128 or higher dimension vector and you could track it (probably similar way you track objects). Associate face and palm tracks per person (possible if openpose is used for both palm & face keypoint generation). This require lot of computation power. You also need to address corner case like when face is partially facing the camera.
